File: src/main/java/de/florianschmitt/service/StringTemplateService.java
package de.florianschmitt.service;

import de.florianschmitt.model.entities.EAbstractUser;
import de.florianschmitt.model.entities.ERequest;
import de.florianschmitt.model.entities.EVoucher;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.stringtemplate.v4.ST;

@Service
public class StringTemplateService {

    @Autowired
    private LinkGeneratorService linkGeneratorService;

    @Value("${application.mail.sender:Organisation}")
    private String sender;

    public String replace(String template, EVoucher voucher, EAbstractUser to, ERequest request) {
        ST st = new ST(template);
        st.add("to", to);
        st.add("request", request);
        st.add("voucher", voucher);
        st.add("sender", sender);
        st.add("linkAnswerRequest", linkGeneratorService.requestAnswer(voucher));
        st.add("linkAnswerRequestDecline", linkGeneratorService.requestAnswerDecline(voucher));
        String result = st.render();
        return result;
    }
}

File: src/main/kotlin/de/florianschmitt/service/StringTemplateService.kt
package de.florianschmitt.service

import de.florianschmitt.model.entities.EAbstractUser
import de.florianschmitt.model.entities.ERequest
import de.florianschmitt.model.entities.EVoucher
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.stringtemplate.v4.ST

@Service
class StringTemplateService {

    @Autowired
    private lateinit var linkGeneratorService: LinkGeneratorService

    @Value("\${application.mail.sender:Organisation}")
    private val sender: String? = null

    fun replace(template: String, voucher: EVoucher, recipient: EAbstractUser, request: ERequest): String {
        val st = replaceCommonVariables(template, recipient.germanDisplayString, request)

        st.add("voucher", voucher)
        st.add("linkAnswerRequest", linkGeneratorService.requestAnswer(voucher))
        st.add("linkAnswerRequestDecline", linkGeneratorService.requestAnswerDecline(voucher))

        return st.render()
    }

    fun replace(template: String, recipient: String, request: ERequest): String {
        val st = replaceCommonVariables(template, recipient, request)

        return st.render()
    }

    private fun replaceCommonVariables(template: String, recipient: String, request: ERequest): ST {
        val st = ST(template)
        st.add("recipient", recipient)
        st.add("request", request)
        st.add("sender", sender)
        st.add("linkCancelRequest", linkGeneratorService.requestCancel(request))
        return st
    }
}
